ui/components/multichain/app-footer/app-footer.js

```diff
@@ -27,113 +30,177 @@ import {
   TextColor,
   TextVariant,
 } from '../../../helpers/constants/design-system';
+import { getSendStage, SEND_STAGES } from '../../../ducks/send';
+import { getUnapprovedTransactions } from '../../../selectors';
 import { showSelectActionModal } from './app-footer-actions';
 
 export const AppFooter = () => {
   const t = useI18nContext();
   const location = useLocation();
   const dispatch = useDispatch();
+  const history = useHistory();
 
-  const walletRoute = `#${DEFAULT_ROUTE}`;
-  const connectedRoute = `#${CONNECTED_ROUTE}`;
   const activeWallet = location.pathname === DEFAULT_ROUTE;
-  const activeConnections = location.pathname === CONNECTED_ROUTE;
+  const activeConnections = location.pathname === CONNECTIONS;
+  const isUnlocked = useSelector((state) => state.metamask.isUnlocked);
+
+  // Disable the network and account pickers if the user is in
+  // a critical flow
+  const sendStage = useSelector(getSendStage);
+  const isTransactionEditPage = [
+    SEND_STAGES.EDIT,
+    SEND_STAGES.DRAFT,
+    SEND_STAGES.ADD_RECIPIENT,
+  ].includes(sendStage);
+  const isConfirmationPage = Boolean(
+    matchPath(location.pathname, {
+      path: CONFIRM_TRANSACTION_ROUTE,
+      exact: false,
+    }),
+  );
+  const isSwapsPage = Boolean(
+    matchPath(location.pathname, { path: SWAPS_ROUTE, exact: false }),
+  );
+  const isSwapsBuildQuotePage = Boolean(
+    matchPath(location.pathname, { path: BUILD_QUOTE_ROUTE, exact: false }),
+  );
+
+  const unapprovedTransactions = useSelector(getUnapprovedTransactions);
+
+  const hasUnapprovedTransactions =
+    Object.keys(unapprovedTransactions).length > 0;
+
+  const hideFooter =
```

Comment:
I think this one is super important, and worth some tests to make sure it is hidden, wdyt?

Reply:
Moved it to routes to keep it same as the header visibility